src/passes/OptimizeInstructions.cpp

```diff
@@ -1259,8 +1261,26 @@ struct OptimizeInstructions
         matches(curr, binary(Abstract::And, pure(&left), ival(0)))) {
       return right;
     }
+    // (signed)x % C_pot == 0   ==>  x & (C_pot - 1) == 0
+    // (signed)x % C_pot != 0   ==>  x & (C_pot - 1) != 0
+    if (matches(curr,
+                binary(Abstract::Eq,
+                       binary(&inner, Abstract::RemS, any(&left), constant(&c)),
+                       ival(0))) ||
+        matches(curr,
+                binary(Abstract::Ne,
+                       binary(&inner, Abstract::RemS, any(&left), constant(&c)),
+                       ival(0)))) {
+      curr->dump();
+      abort();
```

Comment:
@tlively Try new matching api. Could you help me with that? I can't understand why this not matched

Reply:
Alright. Found we need doe this also for unary `EqZ`